### PR TITLE
Add setting to toggle enabling API Cache for Jetpack sites

### PR DESCRIPTION
**Pull request by gravityrail**
This enables the experimental API Cache feature for Jetpack 4.5

See: https://github.com/Automattic/jetpack/pull/6004

### CODE REVIEW

File: client/my-sites/site-settings/form-general.jsx [hunk 1]
```diff
@@ -653,6 +653,34 @@ class SiteSettingsFormGeneral extends Component {
 		);
 	}
 
+	renderApiCache() {
+		// Note that years and months below are zero indexed
```
Comment (lezama) [inline client/my-sites/site-settings/form-general.jsx]: holidaySnowOption :)

File: client/my-sites/site-settings/form-general.jsx [hunk 2]
```diff
@@ -653,6 +656,47 @@ class SiteSettingsFormGeneral extends Component {
...
+		const { fields, translate, isRequestingSettings } = this.props;
+			// today = moment(),
```
Comment (lezama) [inline client/my-sites/site-settings/form-general.jsx]: :)

File: client/my-sites/site-settings/form-general.jsx [hunk 3]
```diff
@@ -653,6 +656,47 @@ class SiteSettingsFormGeneral extends Component {
...
+							'Use synchronized data to boost performance'
+						) }
+					</span>
+				</FormToggle>
+			</CompactCard>
+		);
+	}
+
+	showApiCacheCheckbox() {
+		if ( ! config.isEnabled( 'jetpack/sync-panel' ) ) {
```
Comment (lezama) [inline client/my-sites/site-settings/form-general.jsx]: let's put it under a new feature flag

**Comment by lezama** [comment]
😻

**Comment by gravityrail** [comment]
Addressed @lezama's feedback

**Pull request merged**